src/analytics/analytics-tracker-window.ts

```diff
@@ -1,9 +1,60 @@
+import { isFunction } from 'lodash';
+
 export interface AnalyticsTracker {
     track(step: string, data: any): void;
-    hit(hitName: string, data: any): void;
-    hasPayloadLimit(data: any): boolean;
 }
 
 export default interface AnalyticsTrackerWindow extends Window {
     analytics: AnalyticsTracker;
+    ga(command: string, eventName: string, payload: any): void;
 }
+
+export const sendGoogleAnalytics = (type: string, payload: any) => {
+    const _window = window as unknown as AnalyticsTrackerWindow;
+
+    if (isFunction(_window?.ga)) {
+        _window.ga('send', type, {
+            ...payload,
+            nonInteraction: false,
+        });
+    }
+};
+
+export const isGoogleAnalyticsAvailable = () => {
+    const _window = window as unknown as AnalyticsTrackerWindow;
+
+    return isFunction(_window?.ga);
+};
+
+/**
+ * Max size of the payload for the Google Analytics module
+ * if the limit will be succeeded, the GA throwing a silent error,
+ * and only in debug mode you can see it
+ */
+export const ANALYTICS_MAX_URI_LENGTH = 8096;
+
+export const isPayloadSizeLimitReached = (obj: any) => {
+    return serializeAnalyticsEventPayload(obj).length >= ANALYTICS_MAX_URI_LENGTH;
+};
+
+const serializeAnalyticsEventPayload = (obj: Record<string, any>): string => {
+    return Object.keys(obj).reduce((acc: string[], key) => {
+        const type: string = typeof obj[key];
+
+        if (type === 'string' || type === 'number') {
+            return [
+                ...acc,
+                `${key}=${obj[key]}`,
+            ];
+        }
+
+        if (type === 'object') {
```

Comment:
`null` is also an `object`. You might want to guard against that.

Reply:
good point, done